src/seedu/addressbook/AddressBook.java

```diff
@@ -486,8 +486,12 @@ private static String getMessageForPersonsDisplayedSummary(ArrayList<String[]> p
         final ArrayList<String[]> matchedPersons = new ArrayList<>();
         for (String[] person : getAllPersonsInAddressBook()) {
             final Set<String> wordsInName = new HashSet<>(splitByWhitespace(getNameFromPerson(person)));
-            if (!Collections.disjoint(wordsInName, keywords)) {
-                matchedPersons.add(person);
+            for(String word: wordsInName) {
+            	for(String key: keywords){
+            		if(word.equalsIgnoreCase(key)) {
```

Comment:
indentation is off here!
Our coding guidelines require using 4 spaces for indentation. 
Tip: You can make Eclipse do it for you; check out how!